data-prepper-plugins/kafka-plugins/src/main/java/org/opensearch/dataprepper/plugins/kafka/producer/KafkaCustomProducer.java

```diff
@@ -143,13 +144,15 @@ public Integer getMaxRequestSize() {
     public void produceRecords(final Record<Event> record) throws Exception {
         bufferedEventHandles.add(record.getData().getEventHandle());
         Event event = getEvent(record);
-        final String key = event.formatString(kafkaProducerConfig.getPartitionKey(), expressionEvaluator);
+        final String key = StringUtils.isEmpty(kafkaProducerConfig.getPartitionKey()) ? 
+            null : 
+            event.formatString(kafkaProducerConfig.getPartitionKey(), expressionEvaluator);
         try {
-            if (Objects.equals(serdeFormat, MessageFormat.JSON.toString())) {
+            if (serdeFormat.equalsIgnoreCase(MessageFormat.JSON.toString())) {
```

Comment:
We should not be so lenient. It can allow pipelines with invalid configurations that we don't want to support. e.g. `Json`.

Instead, update `MessageFormat` to expose the `type` value.

```
public String getTypeName() {
  return type;
}
```

Additionally, this method should have a unit test.

Here is an example you can follow.

https://github.com/opensearch-project/data-prepper/blob/af5dce7cd019a577aa6ae64a8081cd9beb787dc2/data-prepper-plugins/aggregate-processor/src/test/java/org/opensearch/dataprepper/plugins/processor/aggregate/actions/OutputFormatTest.java#L47-L61